back/app/models.py

```diff
@@ -62,4 +66,75 @@ def from_dict(cls, data):
             return cls(name=name, age=age, level=level, msg=msg, important=important, status=status, _id=_id)
         except Exception as e:
             raise ValueError(f"Error creating Data object from dict: {e}")
+        
+    @staticmethod
+    def get_statistics():
+        all_records = db.data.find({})
+        records_len = len(list(all_records.clone()))
+        avg_age=0
+        level_distribution={
+            "easy": 0,
+            "normal": 0,
+            "hard": 0
+            }
+        level_distribution_total = 0
+        important_frequency = {True:0,False:0}
+        status_frequency = {"open":0,"closed":0}
+        common_words = defaultdict(int)
+
+        for record in all_records:
+            #calculate avg_age    
+            a = datetime.strptime(record['age'], "%Y-%m-%d").date()
+            b = datetime.now().date()
+            delta = b - a
+            avg_age += delta.days
+
+            #calculate level_distribution
+            level_distribution[record['level']] +=1
+            level_distribution_total +=1
+
+            #calculate important_frequency
+            important_frequency[record['important']] +=1
+
+            #calculate status_frequency
+            status_frequency[record['status']] +=1
+
+            #calculate common_words
+            message = str(record['msg'])
+            message.translate(str.maketrans('', '', string.punctuation)) #remove all punctuation
+            re.sub(' +', ' ', message) # remove multiple spaces e.g 'The   fox jumped   over    the log.'
+
+            for word in message.split():
+                common_words[word.title()] +=1
+
+        #safe gurad to prevent division by zero
+        records_len = max(records_len,1)
+        level_distribution_total = max(level_distribution_total,1)
+
+        #calculating percentages
+        avg_age *= 100/records_len
+        avg_age = round(avg_age,1)
```

Comment:
unattended behavior for age. calculate the average date from all the dates issued in the system.
2020-04-01
2020-04-02
2020-04-03

then the average days is 2 days is the distribution of work hours